ignite/chainconfig/v1/config.go

```diff
@@ -0,0 +1,366 @@
+	OutputDocument string `yaml:"output-document"`
+
+	// PubKey is the validator's Protobuf JSON encoded public key.
+	PubKey string `yaml:"pubkey"`
+
+	// Sequence is the sequence number of the signing account (offline mode only).
+	Sequence uint `yaml:"sequence"`
+
+	// SignMode is the choose sign mode (direct|amino-json), this is an advanced feature.
+	SignMode string `yaml:"sign-mode"`
+
+	// TimeoutHeight sets a block timeout height to prevent the tx from being committed past a certain height.
+	TimeoutHeight uint `yaml:"timeout-height"`
+}
+
+// Validator holds info related to validator settings.
+type Validator struct {
+	// Name is the name of the validator.
+	Name string `yaml:"name"`
+
+	// Bonded is how much the validator has staked.
+	Bonded string `yaml:"bonded"`
+
+	// App overwrites appd's config/app.toml configs.
+	App map[string]interface{} `yaml:"app,omitempty"`
+
+	// Config overwrites appd's config/config.toml configs.
+	Config map[string]interface{} `yaml:"config,omitempty"`
+
+	// Client overwrites appd's config/client.toml configs.
+	Client map[string]interface{} `yaml:"client,omitempty"`
+
+	// Home overwrites default home directory used for the app
+	Home string `yaml:"home,omitempty"`
+
+	// KeyringBackend is the default keyring backend to use for blockchain initialization
+	KeyringBackend string `yaml:"keyring-backend,omitempty"`
+
+	// Gentx overwrites appd's config/gentx.toml configs.
+	Gentx *Gentx `yaml:"gentx,omitempty"`
+}
+
+// GetGRPC returns the GRPC address.
+func (v *Validator) GetGRPC() string {
+	return getConfigValue(v.App, "grpc.address")
+}
+
+// GetGRPCWeb returns the GRPC web address.
+func (v *Validator) GetGRPCWeb() string {
+	return getConfigValue(v.App, "grpc-web.address")
+}
+
+// GetAPI returns the API address.
+func (v *Validator) GetAPI() string {
+	return getConfigValue(v.App, "api.address")
+}
+
+// GetProf returns the Prof address.
+func (v *Validator) GetProf() string {
+	return getConfigValue(v.Config, "pprof_laddr")
+}
+
+// GetP2P returns the P2P address.
+func (v *Validator) GetP2P() string {
+	return getConfigValue(v.Config, "p2p.laddr")
+}
+
+// GetRPC returns the RPC address.
+func (v *Validator) GetRPC() string {
+	return getConfigValue(v.Config, "rpc.laddr")
+}
+
+func (v *Validator) setDefaultAddresses() {
+	if v.App == nil {
+		v.App = make(map[string]interface{})
+	}
+
+	v.App["grpc"] = map[string]interface{}{"address": xnet.AnyIPv4Address(GRPCPort)}
+	v.App["grpc-web"] = map[string]interface{}{"address": xnet.AnyIPv4Address(GRPCWebPort)}
+	v.App["api"] = map[string]interface{}{"address": xnet.AnyIPv4Address(APIPort)}
+
+	if v.Config == nil {
+		v.Config = make(map[string]interface{})
+	}
+
+	v.Config["rpc"] = map[string]interface{}{"laddr": xnet.AnyIPv4Address(RPCPort)}
+	v.Config["p2p"] = map[string]interface{}{"laddr": xnet.AnyIPv4Address(P2PPort)}
+	v.Config["pprof_laddr"] = xnet.AnyIPv4Address(PProfPort)
+}
```

Comment:
@ilgooz regarding your comment about the default config addresses, this is where they are set. The original implementation was assigning default values so I kept the behaviour. Should I remove these default values to keep the migrated config file cleaner?

Reply:
Answering myself I will remove the default values which is the right behaviour.